src/AWSS3.jl

```diff
@@ -1033,11 +1040,34 @@ function s3_complete_multipart_upload(
         upload["Bucket"], upload["Key"], upload["UploadId"], args; aws_config=aws, kwargs...
     )
 
-    return parse(response)
+    return return_raw ? response : parse(response)
 end
 
+"""
+    s3_multipart_upload(aws::AbstractAWSConfig, bucket, path, io::IO, part_size_mb=50;
+                        return_raw::Bool=false, kwargs...)
+
+Upload `data` at `path` in `bucket` using a [multipart upload](https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html)
+
+# API Calls
+
+- [`CreateMultipartUpload`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateMultipartUpload.html)
+- [`UploadPart`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPart.html)
+- [`CompleteMultipartUpload`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CompleteMultipartUpload.html)
+
+# Optional Arguments
+- `part_size_mb`: maximum size per uploaded part, in bytes.
+- `return_raw`: when `true`, return un-parsed (raw) `S3.put_object` response
+- `kwargs`: additional kwargs passed through into `s3_upload_part` and `s3_complete_multipart_upload`
```

Comment:
I've been trying to put the API calls and permissions at the end of the docstring as most readers don't care.
```suggestion
# Optional Arguments
- `part_size_mb`: maximum size per uploaded part, in bytes.
- `return_raw`: when `true`, return un-parsed (raw) `S3.put_object` response
- `kwargs`: additional kwargs passed through into `s3_upload_part` and `s3_complete_multipart_upload`

# API Calls

- [`CreateMultipartUpload`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateMultipartUpload.html)
- [`UploadPart`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPart.html)
- [`CompleteMultipartUpload`](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CompleteMultipartUpload.html)
```